src/policy/policy.cpp

```diff
@@ -231,6 +231,25 @@ bool IsWitnessStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
                     return false;
             }
         }
+
+        // Check P2TR standard limits
+        if (witnessversion == 1 && witnessprogram.size() == WITNESS_V1_TAPROOT_SIZE && !prevScript.IsPayToScriptHash()) {
+            // Taproot spend
+            const auto& stack = tx.vin[i].scriptWitness.stack;
+            size_t stack_size = stack.size();
+            if (stack_size >= 2 && !stack[stack_size - 1].empty() && stack[stack_size - 1][0] == ANNEX_TAG) {
```

Comment:
Is there a name for this condition that can be explained with a function name or at least a comment?

Reply:
I don't see the problem, personally.

Reply:
hey kalle, please see my comment below to pieter. I'm seeing if people are open to using a more consistent single level of abstraction to make the source easier to read.

e.g. this can be evaluated by a newb (ignore if I got the concepts wrong, I'm trying to illustrate a more literate / single level of abstraction style as per http://principles-wiki.net/principles:single_level_of_abstraction)

```
if (payToTapRoot)
  if annexOnStack
    stack_size--; // ignore annex
  else 
    if (OpWhateverOnStack)
      checkWhateverConditionWithLoop
    end
```

Again I could have totally gotten the details wrong but if the program looked like that at a single level of abstraction, with magic numbers given names and conditionals given names, many more people could actually participate in the PR process and grok what's going on.

Reply:
I've added some comments. Better now?

Reply:
Thanks definitely helps, though I would still love to see some vars extracted to named for readability and deduplication. I'm not gonna nitpick your code for style, I'm just a humble bitcoin pleb. But I do know what it's like to scale codebases to be friendly to onboarding new devs. I do think readability should be a concern if we want more devs in bitcoin. e.g. `stack[stack_size - 1]` is repeated all over the place. call it `top_of_stack` and so much more readable! 🤷‍♂ 😄

Reply:
Well the reason is that the `stack_size` variable changes, so a "top_of_stack" one would need to be updated, kinda defeating its meaning.

Reply:
I've changed this code quite a bit. What do you think?